app/scripts/views/sign_in.js

```diff
@@ -37,6 +37,28 @@ define(function (require, exports, module) {
 
     beforeRender: function () {
       this._account = this._suggestedAccount();
+
+      // Extract email to see if valid account
+      var relierEmail = this.relier.get('email');
+      var formEmail = this._formPrefill.get('email');
+
+      // Use the relier's email to check for valid account.
+      // If the form email has been set, this signals that the user has modified the form email input
+      // and we want to use it for later updates
+      if (relierEmail && ! formEmail) {
+        var self = this;
+        this.fxaClient.checkAccountByEmailExists(relierEmail)
```

Comment:
We are moving away from calling the fxaClient directly, instead we are adding corresponding methods to [account.js](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/models/account.js). This has helped us shift a lot of incidental complexity out of views and consolidate places where the fxaClient is referenced directly.